lmcache/address_manager/disk_address_manager.py

```diff
@@ -0,0 +1,280 @@
+                            kv_size: float) -> Iterable[str]:
+        paths = []
+        for key_str in key_strs:
+            key = CacheEngineKey.from_string(key_str)
+            if key in self.dict:
+                kv_size = kv_size - self.dict[key].size
+                paths.append("")
+            else:
+                self.dict[key] = DiskCacheMetadata("", 0)
+                paths.append(self._key_to_path(key))
+
+        evict_keys, put_status = self.evictor.update_on_put(
+            self.dict, DiskCacheMetadata("", kv_size))
+
+        # Abort put if cache too big
+        if put_status == PutStatus.ILLEGAL:
+            # print("Illegal")
+            return [""] * len(key_strs)  # type: ignore
+
+        # evict caches
+        for evict_key in evict_keys:
+            self.remove(evict_key)
+
+        return paths
+
+    def write_ready(self, key_str: str, kv_size: float):
+        """
+        When backend has finished writing the cache, 
+            it will call this function to update the cache 
+            from ("",0) to the actual path and size
+        """
+        key = CacheEngineKey.from_string(key_str)
+        self.dict[key] = DiskCacheMetadata(self._key_to_path(key), kv_size)
+        return True
+
+    def batched_write_ready(self, key_strs: Iterable[str],
+                            kv_sizes: Iterable[float]):
+        for key_str, kv_size in zip(key_strs, kv_sizes):
+            key = CacheEngineKey.from_string(key_str)
+            self.dict[key] = DiskCacheMetadata(self._key_to_path(key), kv_size)
+        return True
+
+    def read_check(self, key_str: str) -> str:
+        """
+        return the path if the key is in the cache, 
+            "" otherwise, including cache that is not ready
+        """
+        self.update_lock.acquire()
+        key = CacheEngineKey.from_string(key_str)
+        if key not in self.dict:
+            self.update_lock.release()
+            return ""
+
+        self.evictor.update_on_get(key, self.dict)
+
+        self.update_lock.release()
+        return self.dict[key].path
+
+    def batched_read_check(self, key_strs: Iterable[str]) -> Iterable[str]:
+        paths = []
+        self.update_lock.acquire()
+        for key_str in key_strs:
+            key = CacheEngineKey.from_string(key_str)
+            if key not in self.dict:
+                paths.append("")
+                continue
+
+            self.evictor.update_on_get(key, self.dict)
+            paths.append(self.dict[key].path)
+        self.update_lock.release()
+        return paths
+
+    def clear(self):
+        self.update_lock.acquire()
+        # Remove all files
+        for key in self.dict:
+            os.remove(self.dict[key].path)
+        self.dict.clear()
+        self.update_lock.release()
+        return True
+
+    # def Info(self):
+    #     return self.info
+
+    def close(self):
+        pass
+
+    def __del__(self):
+        self.close()
```

Comment:
The last three functions seem redundant

Reply:
Info is useful for future usage. It broadcast the manager's config to all the backends so that every KV Cache is stored in an uniformed way.
For the other two function, I could remove them. (I'm not sure about it, please give some advises.)